public/app/features/alerting/unified/utils/templates.ts

```diff
@@ -12,3 +14,9 @@ export function ensureDefine(templateName: string, templateContent: string): str
   }
   return content;
 }
+export function updateDefinesWithUniqueValue(templateContent: string): string {
+  const getNewValue = (match_: string, originalDefineName: string) => {
```

Comment:
non-blocking: I think if I made a copy of a copy it would keep appending `_NEW_0123456789` and would result in a pretty long template name (not sure if there is a limit to it).

Reply:
I think users are not going to keep these suggested names. They probably will update this define values with their particular templates names. The idea of creating this unique define new values was to ensure users could save the new copied template, and not overwrite the old templates.